sql/core/src/main/scala/org/apache/spark/sql/execution/SparkSqlParser.scala

```diff
@@ -713,13 +714,18 @@ class SparkSqlAstBuilder(conf: SQLConf) extends AstBuilder(conf) {
         }
         (Seq.empty, Option(name), props.toSeq, recordHandler)
 
-      case null =>
+      case null if conf.getConf(CATALOG_IMPLEMENTATION).equals("hive") =>
         // Use default (serde) format.
         val name = conf.getConfString("hive.script.serde",
           "org.apache.hadoop.hive.serde2.lazy.LazySimpleSerDe")
         val props = Seq("field.delim" -> "\t")
         val recordHandler = Option(conf.getConfString(configKey, defaultConfigValue))
         (Nil, Option(name), props, recordHandler)
+
+      // SPARK-32106: When there is no definition about format, we return empty result
+      // to use a built-in default Serde in SparkScriptTransformationExec.
+      case null =>
+        (Nil, None, Seq.empty, None)
```

Comment:
For your confuse https://github.com/apache/spark/pull/29085#discussion_r456974391,

CalenderIntervalType/ArrayType/MapType/StructType as input of hive default serde will throw error, won't throw error in spark default way.

Reply:
If so, why does the tests pass in `HiveScriptTransformationSuite`? 
https://github.com/apache/spark/pull/29085#discussion_r456974391

Reply:
> If so, why does the tests pass in `HiveScriptTransformationSuite`?
> [#29085 (comment)](https://github.com/apache/spark/pull/29085#discussion_r456974391)

Since it directly build SparkPlan, don't use sql parser

Reply:
> CalenderIntervalType/ArrayType/MapType/StructType as input of hive default serde will throw error

btw, we already have end-2-end tests for the unspported cases in the hive side?

Reply:
> > CalenderIntervalType/ArrayType/MapType/StructType as input of hive default serde will throw error
> 
> btw, we already have end-2-end tests for the unspported cases in the hive side?

Added